labware-library/src/labware-creator/components/ConditionalLabwareRender.tsx

```diff
@@ -19,74 +20,107 @@ interface Props {
   definition: LabwareDefinition2 | null
 }
 
-const calculateViewBox = (args: {
-  bBox: DOMRect | undefined
-  xDim: number
-  yDim: number
-}): string => {
-  const { bBox, xDim, yDim } = args
-
+const calculateViewBox = (bBox: DOMRect): string => {
   // by-eye margin to make sure there is no visual clipping
   const MARGIN = 5
 
   // calculate viewBox such that SVG is zoomed and panned with the bBox fully in view,
   // in a "zoom to fit" manner, plus some visual margin to prevent clipping
-  return `${(bBox?.x ?? 0) - MARGIN} ${(bBox?.y ?? 0) - MARGIN} ${
-    xDim + MARGIN * 2
-  } ${yDim + MARGIN * 2}`
+  const x = bBox.x - MARGIN
+  const y = bBox.y - MARGIN
+  const xDimension = bBox.width + MARGIN * 2
+  const yDimension = bBox.height + MARGIN * 2
+  return `${x} ${y} ${xDimension} ${yDimension}`
+}
+
+const areBBoxesEqual = (
+  a: DOMRect | undefined,
+  b: DOMRect | undefined
+): boolean => {
+  return isEqual(
+    {
+      x: a?.x,
+      y: a?.y,
+      width: a?.width,
+      height: a?.height,
+    },
+    {
+      x: b?.x,
+      y: b?.y,
+      width: b?.width,
+      height: b?.height,
+    }
+  )
 }
 
 export const ConditionalLabwareRender = (props: Props): JSX.Element => {
+  const { definition } = props
+  return definition === null ? (
+    <Placeholder />
+  ) : (
+    <PopulatedPreview definition={definition} />
+  )
+}
+
+const PopulatedPreview = (props: {
+  definition: LabwareDefinition2
+}): JSX.Element => {
   const { definition } = props
   const gRef = useRef<SVGGElement>(null)
   const [bBox, updateBBox] = useState<DOMRect | undefined>(
-    gRef.current ? gRef.current.getBBox() : undefined
+    gRef.current?.getBBox()
   )
 
   // In order to implement "zoom to fit", we're calculating the desired viewBox based on getBBox of the child.
   // So we have to actually render the child to get its bounding box. After that, we re-calculate the viewBox.
   // Once the viewBox is re-calculated, we use setState to force a re-render.
-  const nextBBox = gRef.current?.getBBox()
   useLayoutEffect((): void => {
-    if (
-      nextBBox != null &&
-      (nextBBox.width !== bBox?.width || nextBBox.height !== bBox?.height)
-    ) {
-      updateBBox(nextBBox)
-    }
-  }, [bBox?.height, bBox?.width, nextBBox])
-
-  const xDim =
-    definition != null
-      ? bBox?.width ?? definition.dimensions.xDimension
-      : DEFAULT_X_DIMENSION
-  const yDim =
-    definition != null
-      ? bBox?.height ?? definition.dimensions.yDimension
-      : DEFAULT_Y_DIMENSION
+    const nextBBox = gRef.current?.getBBox()
+    if (!areBBoxesEqual(bBox, nextBBox)) updateBBox(nextBBox)
```

Comment:
```suggestion
    if (!areBBoxesEqual(bBox, nextBBox)) {
    updateBBox(nextBBox)
    }
```

Always braces when we can. It prevents the occasional odd bug, and after staring at javascript long enough, it actually becomes harder to read `if` statements without them 😄